### ISSUE TITLE
Games with Valve's CEG DRM won't launch

### ISSUE THREAD

**Issue by tfaller** [ValveSoftware/Proton#753]
Already confirmed games which use CEG and won't launch:

Mafia II (50130) #330 
Hitman: Absolution (203140) #282 
Sniper Elite V2 (63380) #392

**Comment by vanyasem** [comment]
Still won't launch with proton 3.16-1

**Comment by ghost** [comment]
Of interest here... Alice Madness Returns is listed as using CEG but launches just fine. https://steamdb.info/app/19680/config/
> "cegpublickey | 3081..."

It does have an EA activation screen however, but that still means that CEG isn't itself a problem unless some games are listed incorrectly or the EA activation worked around it.

Risen 2 is listed as having CEG and fails to start with no error to go by.
Lara Croft and the Guardians of light also listed as having CEG and fails to start.

So from the above it looks like CEG games can run fine but perhaps a check fails with Proton unless the game has some other activation. Its either that or CEG is a coincidence.

I'm still not sold on CEG being the problem here unless the EA activation works around it and allows the game to start, but I do not know the technical ability of the code here - if that's possible for it to be tied into it or not.

The only technical thing I've found so far is the following https://partner.steamgames.com/doc/sdk/uploading

> ""Error code 15" at launch time
> This is the CEG servers rejecting the request to CEG the exe. Check the release state on the game page. If it is not 'playable', you will need to request Steam keys that override the release state."

This suggests only that a game [needs to be set] as playable or has keys that override that state.

> "The Steam Servers are too busy to handle your request... Error Code (2)" at launch time
> This is the CEG servers failing to find a matching CEG exe. Double check that you have a working CEG executable on the CEG tab of your app admin. Click the 'status' button to make sure that it's propped."

This suggests only that a matching CEG exe [needs to be] found.

Both of these shouldn't be the problem here. While is does seem that Valve's servers are rejecting CEG with Proton, again there's no log entry about it that I've seen so no proof that says this is what's happening [except looking like it].

**Comment by Corben78** [comment]
My experiements with AvP indicated that at least for AvP ceg seems to matter. I've posted my results in https://github.com/ValveSoftware/Proton/issues/530#issuecomment-431513884.
Downloading the executables via wine for those games, where proton fails might help.

**Comment by d10sfan** [comment]
I tried that experiment with Sniper Elite V2, and it crashed right after I clicked play.

**Comment by pchome** [comment]
> Downloading the executables via wine for those games, where proton fails might help.

Tested on #563 , this indeed helped a bit: the game still not launching, but now I can see at least `lsteamclient.dll` and `steam.dll` are loaded, and Steam's `[API loaded no]` message in the log file.

**Comment by supertin** [comment]
For Hitman Absolution, I can confirm the DRM is the issue... Using a "no-cd" version of the HMA.exe file allows the game to start up. Not sure if it's still talking to Steam or not, but it runs.

If this is Valve's own DRM, should we expect a Valve fix some time?

**Comment by neuromancer** [comment]
Is this fixed in Proton 4.2?

**Comment by marekjedrzejewski** [comment]
> Is this fixed in Proton 4.2?

Tried Hitman Absolution and it doesn't start.

**Comment by CSahajdacny** [comment]
Is very ironic that a DRM created by Valve doesn't allow us to play games on the Valve's platform.

**Comment by VortexAcherontic** [comment]
> Is this fixed in Proton 4.2?

Nope tested it just a few seconds ago

**Comment by WillHaRSI** [comment]
Verified that Typing of the Dead doesn't start

**Comment by oblitum** [comment]
> > Is this fixed in Proton 4.2?
> 
> Nope tested it just a few seconds ago

My only worry is whether Valve plans to support this in the short term or not, I'm also with a handful of games I can't play on Linux, but gladly I have a bunch of others that work fine, so my plan is to keep playing those for now until this ends up being fixed. I hope it will not take very long.

**Comment by mindinsomnia** [comment]
Reportedly this effects Deadpool as well.
Crossing my fingers a fix for this could be implemented, it seems to effect quite a few quality games. =)

**Comment by XakepSDK** [comment]
I hope next proton release will fix it.
Current proton(for history): 4.2-4

**Comment by GH-A-CC** [comment]
Is anyone at Valve actually looking into this?  Since this stuff was working before (until late 2018?) presumably its an update Valve pushed out somewhere around that time-frame that has broken compatibility with all these games.

I understand these aren't officially supported on Linux, but Valve breaking something that was working perfectly fine for nearly all players, ignoring the issue 7-8+ months, and leaving it up to the community to identify and promote the solution of "get a cracked version" for affected games reflects really, really poorly on the Valve team.

@kisak-valve - is this something that can be escalated internally within Valve to push for a solution?

**Comment by leillo1975** [comment]
> Is very ironic that a DRM created by Valve doesn't allow us to play games on the Valve's platform.

I think exactly the same thing

**Comment by kisak-valve** [comment]
Hello @GH-A-CC, as far as I'm aware, CEG has never worked with Proton and this is not a regression.

Working around this issue with a random binary from an untrusted internet source is never a good answer. Copying the game's CEG binary from a Steam-on-wine install on the same system like is described at https://github.com/ValveSoftware/Proton/issues/530#issuecomment-433696430 is currently the best option to try to workaround this issue and the logs from that method should be valid when troubleshooting the games.

**Comment by CSahajdacny** [comment]
This is another solid evidence that **DRM is BAD** for legit consumers.. 
Illegal users don't have this problem.

There is only one solution to this problem: Valve must get rid of CEG **ASAP**
And don't come to me with the old " there is nothing We can do about it" or " is not that simple"
YES! It is that simple!.. Valve created the CEG DRM, Valve can eliminate the CEG DRM.. End of discussion.

**Comment by GH-A-CC** [comment]
If the issue is actually ..."The Steam Linux client (as well as steamcmd, both win and lin) doesn't download a working executable."... (as per @kisak-valve's reference to [comment #530](https://github.com/ValveSoftware/Proton/issues/530#issuecomment-433696430) above), then presumably this is a bug for the Steam Client team to fix. 

Does anyone know if they have they been informed and are investigating / progressing it?

Edit: 
@mirh (who replied with the confused emoji to this post): If the client is downloading a broken executable, rather than a 'normal' one, it sounds like each game binary using CEG might be customised / tied to a specific Steam account or system at the point the client pulls it down.

**Comment by duderand** [comment]
Please check this issue #2179 .

I think all the issues related with Steam CEG are problems with downloading and replacing the EXE with the correct ones. Besides the timming issue reported on the #2179 I think other issues can be caused by locked EXE files.

Testing with grid2 I can see that wine runs grid2.exe which generates a grid2.exe.12.STEAMSTART file which, is my guess, is Steam API trying to update the EXE for the correct one. That might fail since either Linux or wine is locking the file.

**Comment by pchome** [comment]
@duderand 
> "... which, is my guess, is Steam API trying to update the EXE for the correct one."

Well, I guess it's easy to check with something like `$ inotifywatch -v  grid2.exe`.

**Comment by duderand** [comment]
> Well, I guess it's easy to check with something like `$ inotifywatch -v grid2.exe`.

The theory seems to hold up. Doing `inotifywait -mr "/home/user/.steam/steam/steamapps/common/grid 2/" --timefmt '%F %T' --format '%T %w%f %e'` I get the following log

https://gist.githubusercontent.com/duderand/ff4805ff6f2af3c60902fff63d0857e4/raw/2036aced01725aaaec03f437445d81026cf86799/grid2_fslog.txt

After what seems to be finishing downloading the grid2.exe.39.STEAMSTART it seems to hang for 30 seconds to close everything since it has failed. When it's closing it also closes the grid2.exe file.

I tried to get more info through auditctl but I only get the info what wineserver opens the grid2.exe.. nothing more.

**Comment by duderand** [comment]
Upon further investigation it seems not be a file lock but timeout waiting for STEAM to start :\

Execute grid2.exe #1647 with `WINEDEBUG="+relay,+seh,+tid"` to get more details and found the 30 seconds delay. A thread that downloads the STEAMTART file waits for 30 seconds on a event named `STEAM_START_ACK_EVENT_%Num1%_%Num2%`. After that failing raises the event `STEAM_TERM_EVENT_%Num1%_%Num2%`

Logs 

```
0026:Call KERNEL32.CreateEventA(0032fa3c,00000001,00000000,01c50052 "STEAM_START_ACK_EVENT_37_8899717") ret=00a9e2fd
0026:Ret  KERNEL32.CreateEventA() retval=000000a0 ret=00a9e2fd
0026:Call KERNEL32.CreateEventA(0032fa38,00000001,00000000,01c50073 "STEAM_TERM_EVENT_37_8899717") ret=00a9e2fd
0026:Ret  KERNEL32.CreateEventA() retval=000000a4 ret=00a9e2fd
...
0026:Call KERNEL32.WaitForSingleObject(000000a0,0000752f) ret=0087cc96
...
0026:Ret  KERNEL32.WaitForSingleObject() retval=00000102 ret=0087cc96 <<<- TIMEOUT
0026:Call KERNEL32.CloseHandle(000000a0) ret=0087cca9
0026:Ret  KERNEL32.CloseHandle() retval=00000001 ret=0087cca9
0026:Call KERNEL32.SetEvent(000000a4) ret=005a32e0
0026:Ret  KERNEL32.SetEvent() retval=00000001 ret=005a32e0
0026:Call KERNEL32.CloseHandle(000000a4) ret=005a32e7
0026:Ret  KERNEL32.CloseHandle() retval=00000001 ret=005a32e7
```

Thread 0026 seems to be waiting on thread 0027 which is also waiting on something :\

**Comment by duderand** [comment]
Further testing reaveals that grid2.exe really depends on a running steam.exe application. In other words, for grid2.exe to start a windows steam application must be running.

I checked the logs and checked a steam install on a windows application and the missing pipes and events are present there (setup when STEAM.exe starts) so there's a real dependency (in many games) between the game and a windows steam version.

My next step will be to create a prefix with a steam install there (not sure if possible) so that maybe a game can talk to a windows steam and be happy with it.

**Comment by pchome** [comment]
@duderand 
> Further testing reaveals that grid2.exe really depends on a running steam.exe application. In other words, for grid2.exe to start a windows steam application must be running.

AFAIK [it should be running](https://github.com/ValveSoftware/Proton/search?q=steam.exe&type=Commits) when launched by Proton.

Maybe case sensitive, and should be renamed to `STEAM.exe`?
IIRC Location: `dist/lib/wine/steam.exe.so`

**Comment by duderand** [comment]
> AFAIK [it should be running](https://github.com/ValveSoftware/Proton/search?q=steam.exe&type=Commits) when launched by Proton.
> 
> Maybe case sensitive, and should be renamed to `STEAM.exe`?
> IIRC Location: `dist/lib/wine/steam.exe.so`

That steam.exe seems to be a simplified version of the real steam ("This is a stub steam.exe" [reference](https://github.com/ValveSoftware/Proton/blob/proton_4.11/steam_helper/steam.cpp)). What I'm guessing is that the injected steam in wine prefix doesn't provide the same inter process communication that the windows version (or maybe wine doesn't support part of it). To be more specific I don't think the injected steam.exe or steamservice launches events (kernel api [link](https://docs.microsoft.com/en-gb/windows/win32/api/synchapi/nf-synchapi-setevent)) and doesn't  create named pipes (win32 api [link](https://docs.microsoft.com/en-us/windows/win32/ipc/named-pipes)) that the game requires for the CEG.

**Comment by Shished** [comment]
Civlization V is also affected.

**Comment by Guy1524** [comment]
@duderand I looked into this, CEG requires that steam setup their STEAM_DRM_IPC interface.  Windows steam of course does this, but I'm not sure if Linux steam supports CEG in any way.

I searched the interface on github and found [this bypass software](https://github.com/Hedgehogscience/STMSIGBypass_cpp), which is useful as a starting point for hooking this up with any CEG API that may exist on the Linux client.

**Comment by mirh** [comment]
> Windows steam of course does this, but I'm not sure if Linux steam supports CEG in any way.

Ehrm... That doesn't explain why CEG doesn't work in normal wine steam either.

**Comment by duderand** [comment]
@Guy1524 nice find!

At this point I have stopped working on this issue since it's not a WINE limitation but a limitation on Steam DRM implementation. My guess is that Valve would probably reject a PR that touches their DRM.

Either way, If I understand that repo code they have only implemented CEG integrity checks bypasses. CEG also defines a way for the game executable, with a dependency on Steam CEG API, to self modify. This self-modification is required to start the game the first time and that is probably more complex and requires more reverse engineering that I can endure.

**Comment by duderand** [comment]
> > Windows steam of course does this, but I'm not sure if Linux steam supports CEG in any way.
> 
> Ehrm... That doesn't explain why CEG doesn't work in normal wine steam either.

Some games do like Grid2. 

The games that don't work is caused by either:
1. WINE limitations because either STEAM or the game uses APIs (Windows, .net, DirectX, etc.) that aren't implemented or have bugs
1. Steam CEG, like any DRM, is check the steam/system/game integrity and is detecting issues
    * Since WINE doesn't behave exactly like Windows it's expected that some integrity checks fail. For example a file in windows file system can have different properties than in linux filesystem through WINE.

My bet is on WINE limitations since Steam CEG doesn't seem that much aggressive.

**Comment by mirh** [comment]
> Some games do like Grid2.

Mhh.. I see. 

> For example a file in windows file system can have different properties than in linux filesystem through WINE.

Unless they are precisely checking some dll checksum (and they definitively could not even, since operating systems get updated every now and then) it's not like that still isn't up to wine to get right. 

I have another theory though after noticing the heroic report of the grid 2 guy https://github.com/ValveSoftware/Proton/issues/1647#issuecomment-474460265
The folder `mui` in system32 is one of the many CEG uses to "bind" the executables to the computer. Now, I'm not valve and I don't know which rationale they use to pick up "targets for sealing"... But it could be that a normal wine prefix is usually so sparse that there isn't even have the minimum required number of elements for the algorithm to run. 
(or still, simply enough, whatever trivial info they query from the file system isn't bit-per-bit identical to windows)
If just there were decent process monitors in linux to be able to guess this..

**Comment by VortexAcherontic** [comment]
@duderand For my experience (Duke Nukem Forever for example) CEG only do not work with Proton if you play the same game with vanilla Wine the games work, if there are no other issues.

**Comment by duderand** [comment]
I think the issue with Duke Nukem Forever ( #782 ) is probably not related with CEG but with a proton limitation. The log on #782 seems to show the failure is caused by mss32.dll not being loaded. Tracking that issue on wine db this https://forum.winehq.org/viewtopic.php?t=32230

This leads me to concluded that Duke Nuken Forever doesn't work because the way the Steam and game installation works with Proton. I really don't understand how and what the issue really is but it's not related to CEG, it's just a proton issue.

**Comment by pchome** [comment]
Okay, next round. I played a bit with **The Darkness ||**.

I patched Proton's steam.exe with the code from mentioned "bypass software", was able to to get some data like ID/status/names, but the game still crashed (this time w/o waiting for 30 sec.).

Next, I took executable from Steam client for Windows (under Wine), this time the game started and crashed after all intro videos was played. Also, the status variable value was changed from 0 to 1. Overall data received from IPC was the following:
```
id: 50
status: 1
module: Z:\media\SteamLibrary\SteamApps\common\Darkness II\DarknessII.exe.50.STEAMSTART
start: STEAM_START_ACK_EVENT_50_5771412
stop: STEAM_TERM_EVENT_50_5771412
```
Next, I googled a bit and applied some patches on wineserver ([bug 46472](https://bugs.winehq.org/show_bug.cgi?id=46472) and [bug 32515](https://bugs.winehq.org/show_bug.cgi?id=32515) partially). This time I was able to browse game's menu and play prelude video. **The game now crashing during "new game" start process.**

The game seems also affected by the "time bug", I checked with libfaketime and it just didn't start with FAKETIME=-2d (so "today" and "yesterday" is okay, and the game crashing for an different reason).

I also tried **Grid 2**, but process became unresponsive and remained <defunct> in the process list, so I must manually kill it every time.

Maybe I need to apply staging patches on Proton, or check an known game (e.g. I played Just Cause 3 and Batman: AK under Wine a while ago, so I know they should work in Steam just fine).

@aeikum , @Guy1524 , any chance this information is helpful to proceed further?

**Comment by neuromancer** [comment]
> Next, I googled a bit and applied some patches on wineserver (bug 46472 and bug 32515 partially). This time I was able to browse game's menu and play prelude video. The game now crashing during "new game" start process.

Perhaps @GloriousEggroll can include them in the next version of proton-ge-custom, so we can easily test other CEG games and report back the results here?

**Comment by FanderWasTaken** [comment]
Proton 5.0-1 got released. Changelog says this:

> Improved Steam client integration. This makes more games that use Denuvo playable, including Just Cause 3, Batman: Arkham Knight, Abzu, and more.

Can anyone check if «Improved Steam client integration» did anything for titles with CEG?

**Comment by petr666** [comment]
Unfortunately no change (at least for Typing of The Dead)

**Comment by lucifertdark** [comment]
Warhammer 40,000: Space Marine is another game on the list that still doesn't work without resorting to sketchy executable replacement. tested with Proton 4.11-12 & 5.0-1. I'm currently waiting for Hitman Absolution to install so I can test that as well.

**Comment by lucifertdark** [comment]
I can confirm Hitman Absolution is still affected.
[steam-203140.log](https://github.com/ValveSoftware/Proton/files/4174087/steam-203140.log)

**Comment by pchome** [comment]
> Improved Steam client integration. This makes more games that use **Denuvo** playable, including Just Cause 3, Batman: Arkham Knight, Abzu, and more.

You can expect some of [this](https://www.pcgamingwiki.com/wiki/Denuvo#List_of_games_using_Denuvo_Anti-Tamper) games will now work.

**Comment by mirh** [comment]
> Can anyone check if «Improved Steam client integration» did anything for titles with CEG?

This is the magic probably https://github.com/ValveSoftware/wine/commit/a7858a380032403bac2f52a0245cb28ee6d19592 (and https://github.com/ValveSoftware/wine/commit/f1a778c4f1c854be135b8bd9e320dde98fcca78a)
Which indeed, suggests me something similar would be needed to accommodate CEG.

**Comment by Newbytee** [comment]
> Can anyone check if «Improved Steam client integration» did anything for titles with CEG?

Mafia II still crashes on launch for me.

**Comment by vpereira01** [comment]
Tested Grid 2 and still see the same behavior, timeout because the STEAM_START_ACK_EVENT kernel even is not set.

**Comment by duderand** [comment]
Tested Grid 2 and still see the same behavior, timeout because the STEAM_START_ACK_EVENT kernel even is not set.

**Comment by mathew2214** [comment]
is there a way to manually generate our CEG files to then place them into a game's directory such that it launches with Proton?

**Comment by FanderWasTaken** [comment]
Maybe launching game from Lutris could do that, since games with this DRM
work there. But just maybe 😉

**Comment by mathew2214** [comment]
all i think we need is a method of capturing and permanently storing the generated files. 
from testing, i know CEG doesnt replace the files in the game's installation directory. maybe there's some sort of temp folder CEG puts its generated files?

**Comment by int-72h** [comment]
This appears to be an issue Valve can only tackle, apart from previously mentioned bypass software and the hack from a steam-on-wine install. Could someone automate testing games using these methods so we have a very rough compatibility guide, or are these processes too specific? If so, the trick used may be useful being integrated into Proton in some form, since Valve won't get rid of the DRM.

**Comment by FanderWasTaken** [comment]
Well, @int-72h, I don't really know how to automate this, but at least I installed these games as follows:

1. Install Lutris.
2. Install Steam runner.
3. Search for the game.
4. If it's there install it with a provided script; if it is not there add a new game for Steam runner manually, all needed is the `steamid` of the game.
5. Click Play.

It should load Steam in Wine and then start downloading the game. Or, if you can spend a bit more time, you can add your local steam directory to Wine's Steam's `winecfg` as a partition. This way you can download games in native Steam client and just run them through Wine's, if it doesn't work otherwise. Also shortcut from Lutris to the game can be added to native Steam client, so you could have almost seamless transition between native client and Wine's one.

**Comment by mirh** [comment]
5.0-4 brought in a lot of steamclient/helper work. 
Can you recheck?

**Comment by aeikum** [comment]
Won't help. We need support from the Steam client, but they don't ship CEG support in Steam for Linux. I'll update this bug if there's progress here.

**Comment by leillo1975** [comment]
I hope @ValveSoftware make it possible. There are a lot of great games that uses CEG

**Comment by Eric1212** [comment]
@kisak-valve Did you make sure that others employes of Valve are aware of this issue to be fixed ?

**Comment by tpurde** [comment]
Call of Duty: Black Ops II is also affected by this issue. I hope something will be done.

**Comment by CSahajdacny** [comment]
This issue must be top priority because is Valve's fault.

**Comment by sorin255** [comment]
> This issue must be top priority because is Valve's fault.

Not really because most of these games do have Linux ports so it's not like they're unplayable. They're not as good as they should be but they're not unplayable.

Maybe there could be a list of games whose Linux ports are too broken to be playable and those specific ones could be a priority. Along with games without a port.

**Comment by tpurde** [comment]
@sorin255 What you said is straight up wrong. I checked almost every single game reported in this page and none of them were native. That's why we want this issue fixed, please get your facts right.

**Comment by sorin255** [comment]
I have a bunch of games in my library that have CEG and a native. The issue
I opened for one of them redirected me here. I figured it was a similar
story for most.

**Comment by jqadev** [comment]
@Plagman shared his email here https://twitter.com/Plagman2/status/910307267769131008?s=20
Let's write to him directly on e-mail, otherwise we will never get any answer on GitHub.
I've already kindly asked him to give an answer to my question from https://github.com/ValveSoftware/steam-for-linux/issues/7035

**Comment by Newbytee** [comment]
> @Plagman shared his email here https://twitter.com/Plagman2/status/910307267769131008?s=20
> Let's write to him directly on e-mail, otherwise we will never get any answer on GitHub.
> I've already kindly asked him to give an answer to my question from [ValveSoftware/steam-for-linux#7035](https://github.com/ValveSoftware/steam-for-linux/issues/7035)

No, please don't spam him about this. Granted, I don't know him — but generally it's not very nice manners. Valve are definitely aware of this issue. Plagman commented on CEG support in the VKx Discord multiple times, so he's certainly aware of it.

In case you're curious, he said this:
![image](https://user-images.githubusercontent.com/4428753/79777885-32170c80-8338-11ea-845a-90a37cee3f37.png)

**Comment by tpurde** [comment]
@Newbytee When was this? I hope they actually do it.

**Comment by Newbytee** [comment]
> @Newbytee When was this? I hope they actually do it.

November 1st, 2019. He also said this in the 16th of December, 2018:
![image](https://user-images.githubusercontent.com/4428753/79779071-07c64e80-833a-11ea-860f-31b1310cd600.png)

**Comment by tpurde** [comment]
Damn it. CEG actually works with Wine, though. Can't see why it doesn't work through proton.

**Comment by Newbytee** [comment]
> Damn it. CEG actually works with Wine, though. Can't see why it doesn't work through proton.

It's because in Proton there is no Windows Steam. There's the fake/stub steam.exe to please some games, but nothing to provide CEG (CEG is not available for Linux and thus not part of Steam for Linux). Even if CEG was added to Steam for Linux, you'd need to add some glue code to make this work, and I can only imagine this is more difficult than it sounds.

This is my understanding anyway, don't quote me on this.

**Comment by jqadev** [comment]
Guys who gave me thumb down. It is a publicly available company, not private email address shared by the owner who asked to contact with him :)
Even Gabe Newell shared his email address so that anyone can contact with him.

Some people are waiting for information on progress in this matter, even for any short response, status update for over a year.

Look, thanks to that we have now clear answers pasted by @Newbytee and we don't have to wait any longer for clarification :)

**Comment by tpurde** [comment]
Well, Valve was the one who didn't port the DRM to Linux, I think they should do it. DRM almost isn't a problem at all in Proton, and the only "big" DRM which doesn't work is one developed by Valve, a super Linux positive company.

**Comment by Ruedii** [comment]
CEG is the stronger of Valve's three first-party DRM options.  Few games use it.  They don't even use it themselves.  This is probably why it isn't a priority.  

I noticed the DLLs are there in the Proton install and it is passing the necessary values.  There is a new option in the latest Proton for forcing using proper syscall  methods.  Maybe this will fix it.

**Comment by tpurde** [comment]
@Ruedii Is what you're talking about on a released version of Proton? If so, can you test it?

**Comment by Ruedii** [comment]
> @Ruedii Is what you're talking about on a released version of Proton? If so, can you test it?

I seriously don't think it's enough, it's one of TWO things that need to be done.

The second is expanding the Steam infrastructure in SteamPlay.   However, the reason they weren't bothering adding said infrastructure is it wouldn't have worked anyway without the new setting.

p.s. the new setting is "PROTON_USE_SECCOMP=1" if you want to test it.

I suspect this setting also fixes a few Anti-Cheat systems as well as a few treating bugs as features quirks.  (The big one there is the NASTY self-adapting code hack of checking various aspects of the call stack to determine how and where a routine was called from and executing it differently based on it.)

I've got a lot of games on my plate to test with that new setting before I think of adding new games to my "To Test" list.  However, I do have one game that I can test later if I find the time.  

Update:
I read in another bug that a few CEG titles are now working.  It likely depends on the build profile of CEG.

**Comment by Newbytee** [comment]
> Update:
> I read in another bug that a few CEG titles are now working. It likely depends on the build profile of CEG.

Mafia II is still a no-go. Where did you read that?

**Comment by int-72h** [comment]
Only way CEG is going to work is by valve implementing a better stub .exe through wine, even though they have the source for CEG it may rely on some odd windows specific tricks that will be hard to get around in proton.

**Comment by frostworx** [comment]
At least Mafia 2 and Hitman Absolution have DRM free binaries officially on gog, so they do exist.
shouldn't it be possible for Valve to simply offer those/these binaries? would be the easiest and most obvious fix for this issue. GfWL was patched out of several games as well.

**Comment by Newbytee** [comment]
> At least Mafia 2 and Hitman Absolution have DRM free binaries officially on gog, so they do exist.
> shouldn't it be possible for Valve to simply offer those/these binaries? would be the easiest and most obvious fix for this issue. GfWL was patched out of several games as well.

I think this would be up to the publishers, not Valve.

**Comment by frostworx** [comment]
yeah sure, but it's both Valves platform and issue tracker here, so a valid place to show demand for a fix. pretty sure Valve has more options to talk with the publisher. I'm no lawyer but even without the publisher I'm not sure if it would be a problem to offer official drm free executables for owners of the game.

edit: btw gog mafia2.exe (plus two dlls) work fine with the steam installation.
don't own hitman absolution on gog to test.

**Comment by tpurde** [comment]
> Only way CEG is going to work is by valve implementing a better stub .exe through wine, even though they have the source for CEG it may rely on some odd windows specific tricks that will be hard to get around in proton.

CEG already works in standard Wine through Steam for Windows. That's how I play Black Ops 2. @int-72h

**Comment by mirh** [comment]
It has already been said this happens because proton uses the native linux client as much as possible, and that is missing the CEG infrastructure. 
Can we stop this charade? There's nothing else to add or discover.

**Comment by int-72h** [comment]
> > Only way CEG is going to work is by valve implementing a better stub .exe through wine, even though they have the source for CEG it may rely on some odd windows specific tricks that will be hard to get around in proton.
> 
> CEG already works in standard Wine through Steam for Windows. That's how I play Black Ops 2. @int-72h 

Apologies for my misunderstanding - couldn't you just patch Proton with the standard windows one if it works fine through Wine, or am I missing something?

**Comment by Newbytee** [comment]
> > > Only way CEG is going to work is by valve implementing a better stub .exe through wine, even though they have the source for CEG it may rely on some odd windows specific tricks that will be hard to get around in proton.
> > 
> > 
> > CEG already works in standard Wine through Steam for Windows. That's how I play Black Ops 2. @int-72h
> 
> Apologies for my misunderstanding - couldn't you just patch Proton with the standard windows one if it works fine through Wine, or am I missing something?

As in, you'd run the entire Windows Steam client in Proton? I imagine that would come with its own host of issues as you have 2 clients running then. Better to have the CEG part split off into its own component I would imagine.

**Comment by frostworx** [comment]
before unsubscribing this issue, I'd like to mention that the Old Time Reality Mod (here3.5-123-Beta3-5, f.e. on nexus) for Mafia 2 also contains a mafia2.exe which allows you to play the game on steam via proton. hf!

**Comment by tpurde** [comment]
> > > > Only way CEG is going to work is by valve implementing a better stub .exe through wine, even though they have the source for CEG it may rely on some odd windows specific tricks that will be hard to get around in proton.
> > > 
> > > 
> > > CEG already works in standard Wine through Steam for Windows. That's how I play Black Ops 2. @int-72h
> > 
> > 
> > Apologies for my misunderstanding - couldn't you just patch Proton with the standard windows one if it works fine through Wine, or am I missing something?
> 
> As in, you'd run the entire Windows Steam client in Proton? I imagine that would come with its own host of issues as you have 2 clients running then. Better to have the CEG part split off into its own component I would imagine.

he probably meant patching Proton with CEG.

**Comment by NikoBellicRU** [comment]
> CEG already works in standard Wine through Steam for Windows. That's how I play Black Ops 2. @int-72h

How you can you play bo2, is it cracked? When i open it with wine steam i get "unhandled exception caught" also tried plutonium client and didn't work :(

**Comment by soredake** [comment]
https://steamdb.info/app/50300/config/

**Comment by Newbytee** [comment]
> https://steamdb.info/app/50300/config/

What about this?

**Comment by ghost** [comment]
> > CEG already works in standard Wine through Steam for Windows. That's how I play Black Ops 2. @int-72h
> 
> How you can you play bo2, is it cracked? When i open it with wine steam i get "unhandled exception caught" also tried plutonium client and didn't work :(

Actually CEG games can work if the game's developers implemented an alternative like EA with their login gate which can get it to pass. Just because a game has CEG doesn't mean it can't work.

Adding to that, devs can do whatever they want to their game. They can have it pass CEG if you enter xyzabc123OpenSesame if they wanted.

**Comment by duderand** [comment]
I think some games might work with Proton 5.13 https://github.com/ValveSoftware/Proton/releases/tag/proton-5.13-1b

Since PROTON_USE_SECCOMP has been obsolete that might mean that Proton has a better compatibility with DRM.

**Comment by mathew2214** [comment]
> I think some games might work with Proton 5.13 [`proton-5.13-1b` (release)](https://github.com/ValveSoftware/Proton/releases/tag/proton-5.13-1b)
> 
> Since PROTON_USE_SECCOMP has been obsolete that might mean that Proton has a better compatibility with DRM.

i am still unable to launch CEG games with proton. even with this new update. Valve still doesn't support their own DRM.

**Comment by NikoBellicRU** [comment]
I tried again Saints row 3, Black ops 1/2, MW2 SP and still nothing :/

**Comment by lucifertdark** [comment]
[steam-203140.log](https://github.com/ValveSoftware/Proton/files/5397093/steam-203140.log)
Hitman Absolution is also still affected by the CEG bug.

**Comment by oblitum** [comment]
Folks, no need to keep confirming, I guess we can all assume no effort was made to solve that on this new release.

**Comment by deoxal** [comment]
@byte1024 What do you mean, make it pass? I know what CEG stands for but don't understand how it actually operates.

@Ruedii What are the other 2 types of DRM they have?

**Comment by Corben78** [comment]
With the announcement of the Steam Deck and the announced improvements for Proton, will this issue be addressed as well?

**Comment by mathew2214** [comment]
> With the announcement of the Steam Deck and the announced improvements for Proton, will this issue be addressed as well?

i highly doubt it. if Valve had any intent of Proton supporting CEG, i see no reason why it hasn't been done already. Valve's complete silence on this issue over the past several years gives me the impression they have abandoned all things CEG related. 
i doubt its a technical issue for Valve as raw WINE runs CEG fine. all Valve has to do to fix this is implement a feature-complete steam stub into Proton.

**Comment by CSahajdacny** [comment]
Maybe, when the release of SteamDeck, Valve will tell publishers that this DRM is not supported anymore and all games with CEG will be turned DRMFREE, giving enough time to the publishers to adopt another DRM (like Denuvo) if they want.

**Comment by mathew2214** [comment]
> Maybe, when the release of SteamDeck, Valve will tell publishers that this DRM is not supported anymore and all games with CEG will be turned DRMFREE, giving enough time to the publishers to adopt another DRM (like Denuvo) if they want.

its Valve's DRM. it doesnt have to be unsupported.

**Comment by echaskaris** [comment]
It seems mostly older games use CEG, so they might just do nothing...

**Comment by boltronics** [comment]
I honestly believe Proton is the primary reason why the Steam Deck will likely fail. Valve should be working at 110% capacity to ensure maximum Proton compatibility to avoid a chance of scathing reviews at launch.

Telling people that want to play something like Warhammer 40,000: Space Marine (or any other game on the long list of affected titles) on a Steam Deck to first go and download a crack for Valve's CEG DRM, is not going to end well. I think a lot of people expecting a console-like experience are going to be sorely disappointed. This bug needs to be fixed, and there also likely needs to be some kind of automatic calls to protontricks in the background when launching a Steam game into a new Proton prefix to work-around known issues.

**Comment by mindinsomnia** [comment]
Valve have stated their goal is to implement all Windows APIs and to get all games working between now and the launch of the Steam Deck, and stated work has been completed that has not yet been merged into the public version of Proton.

So I think it would be worth giving Valve the benefit of the doubt until the launch of the Steam Deck. We know they have Steam OS 3.0 internally that's based on Arch, and other stuff we probably haven't seen.

So lets just be patient and see what happens.

**Comment by shigutso** [comment]
Using this [Google search](https://www.google.com/search?q=site:steamdb.info+%22cegpublickey%22+-demo) brings more than 700 results.
That's 700+ games on Steam that will potentially not work on Steam Deck if Valve doesn't address this. Many are AAA.
I checked some of them on ProtonDB and indeed many people report the game not opening at all and in other games people reporting that is due to the CEG DRM.

"Oh but it's only for old games, no one cares."
I've seen many people saying that they will buy the Steam Deck to play their library backlog, so this includes old games that may have this DRM.
Valve said they are working with EAC and BattleEye devs, but CEGDRM is not anti-cheat... any word from them about this would be great.

**Comment by mirh** [comment]
CEG already works with wine, the problem with proton is not windows apis but that the steam native linux client this uses has no support for providing the right files. 
So, please stop to spam.

**Comment by boltronics** [comment]
So the game cache validation issue was fixed? If that's true, mind updating the bug report over at winehq.com? If I'm reading that correctly, it seems to suggest that patches are still required for that part of the solution.

https://bugs.winehq.org/show_bug.cgi?id=32515

It's incredible to think that this CEG DRM has been causing issues in one way or another for GNU/Linux users for close to a decade. I see my name in that bug report from testing games back in 2013.

**Comment by leillo1975** [comment]
No news about this issue?

**Comment by mathew2214** [comment]
> No news about this issue?

I'm pretty sure this is a "wontfix" scenario. given valve's lack of any comment on this issue.

**Comment by CSahajdacny** [comment]
> > No news about this issue?
> 
> I'm pretty sure this is a "wontfix" scenario. given valve's lack of any comment on this issue.

Maybe the fix will be SteamDeck (SteamOS 3.0) exclusive.

**Comment by MathiewMay** [comment]
> > > No news about this issue?
> > 
> > 
> > I'm pretty sure this is a "wontfix" scenario. given valve's lack of any comment on this issue.
> 
> Maybe the fix will be SteamDeck (SteamOS 3.0) exclusive.

nothing will be SteamDeck exclusive lol its open we could just copy and paste the files to a regular linux install.

**Comment by CSahajdacny** [comment]
The only efficient fix for this problem is to remove CEG from the existence...

**Comment by mathew2214** [comment]
> The only efficient fix for this problem is to remove CEG from the existence...

that is unlikely to happen. it'd be much easier for Valve to simply implement a proper Steam stub in Proton. using something like Playonlinux, which makes a proper steam stub, CEG works fine. Playonlinux is open-source. the only excuse for this is Valve being lazy. it would be understandable if Valve came out and said "sorry, we just dont wanna fix it", but this silence is insulting.

**Comment by Newbytee** [comment]
> using something like Playonlinux, which makes a proper steam stub, CEG works fine

It doesn't make a "steam stub". It runs the entirety of Windows Steam through Wine, which is why it works there.

> but this silence is insulting

Plagman has talked about it on Discord and if I recall correctly it comes down to the relatively low number of games that use it and that no upcoming games will use it, and thus it's low priority for them. Furthermore, from what I understand Linux Steam doesn't have CEG implemented at all, so you can't just make the stub Steam DLL Proton provides use CEG code from the native Linux Steam since it doesn't exist there. You'd have to somehow extract the CEG parts of Steam and make that its own thing, which I imagine is harder than it sounds.

If I understand things right anyway.

**Comment by jarrard** [comment]
This seems like something Valve could fix if it already works via the Wine method. Certainly a good idea to resolve the problem before Steam-Deck gets in peoples hands as its a tad embarrassing iyam. :)

**Comment by Ruedii** [comment]
> This seems like something Valve could fix if it already works via the Wine method. Certainly a good idea to resolve the problem before Steam-Deck gets in peoples hands as its a tad embarrassing iyam. :)

There are a few hypothesis I have on no news on this.

I suspect they need to make sure any fix does not weaken the DRM due to agreements with game developers.   

Any fix will likely be a binary tarball as well, due to them likely having NDA with upstream owners of the technology they used in it and with developers that integrate the technology into their games.  

It is very unlikely early versions of such a patch would be visible in the public github for this reason.  Additionally, it is likely they would not disclose any WIP patch until they put it through full debugging for security.

Sadly, if these assumptions are correct, it would mean that short of an actual functioning patch, no news is good news.

Closed with "Won't Fix" would be the bad news.

It is very likely that quite a robust additional stack would be needed to properly link the Proton container to the DRM, and a stack on the Linux side would need to be built to handle the authorization so that it doesn't register as a new machine every time the proton version changes.   This is a substantial amount of work.  

If I was a director valve SteamDeck team, I would backburner this and try instead my first option of phasing out this DRM for their new SteamWorks DRM, which is already compatible with Proton.   I'm sure most of you would take the same approach. 

That said, there may be a more secure version of SteamWorks DRM in the works that may require some additional stacks to work on Linux.  Most likely some form of Trusted Platform crypto vault, or some way to make a 100% replication capable GUID for a system.

**Comment by mindinsomnia** [comment]
I don't think anyone can know for certain what is happening internally at Valve or what kind of obstacles they may be facing on this issue, we can make guesses at best.

What I think we can know for sure is: If Valve could easily fix this issue they would.

Valve have not just demonstrated heavy commitment to this strategy of getting games running via Proton, they are now also **heavily** **invested** in this strategy thanks to the Steam Deck, and highly motivated to get every game on Steam working. Very soon potentially millions of Steam users will be trying to run CEG DRM affected games via Steam Deck. Valve is no doubt quite well aware of this issue, how could they not be?

If there was a quick and easy fix, they would have already employed it.
I doubt Valve will simply write off fixing this or mark it as "Won't Fix", too many popular old games are affected.

Safe to say, this issue will be fixed "as soon as possible", and we will find out how soon that is when it's fixed.

**Comment by smintman** [comment]
Managed to get Just cause 2 working with latest beta and experimental proton. Still had to do some proton fixes for direct x issues. But looks like CEG is fixed. At least on this title?

**Comment by mindinsomnia** [comment]
Well.. the timing of this is fascinating to me. 😛 

(Valve, did you just notice a quick easy fix to this? ;P)

I was able to get Duke Nukem to launch. Wasn't playable but it launched.
And Warhammer 40,000 Space Marine now gets a bit further, I get an error message saying "The Steam servers are too busy to handle your request for Warhammer 40,000: Space Marine Error Code (2)".

**Comment by mathew2214** [comment]
> Managed to get Just cause 2 working with latest beta and experimental proton. Still had to do some proton fixes for direct x issues. But looks like CEG is fixed. At least on this title?

https://steamdb.info/app/8190/config/
doesnt look like JC2 uses CEG.

@mindinsomnia which Duke Nukem game? on DNForever i get that same steam message. 
actually, i now get that message with all CEG games. 
https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3076515048864294055
client beta just today. although it doesnt seem to be fully deployed. 
@kisak-valve can you elaborate on this? the announcement isnt very detailed.

**Comment by Plagman** [comment]
Hello,

Can you please re-test with the new Oct. 29 2021 02:21 Client Beta and the latest Proton Experimental?

**Comment by philipl** [comment]
@Plagman Just tried Transformers: Fall of Cybertron and it launched, for the first time ever. ~~I think it might have some texture problems (extremely low res textures at highest setting), but I assume that's unrelated.~~ (known problem with 4k) Thanks!

**Comment by philipl** [comment]
For Lara Croft and the Guardian of Light, I got the error about steam servers being too busy.

**Comment by d10sfan** [comment]
Just Cause 2 - got black screen then crash after doing the protontricks
Space marine - black screen then crash
Hitman Absolution - works

I put logs in the particular tickets.

**Comment by Patola** [comment]
Lara Croft and the Guardian of Light -- same errors about steam servers being too busy. Will try again later.
The Bureau: XCOM Declassified -- doesn't start, added steam log to the proper bug report.

**Comment by mindinsomnia** [comment]
> 
> 
> Hello,
> 
> Can you please re-test with the new Oct. 29 2021 02:21 Client Beta and the latest Proton Experimental?

Getting there!
Just tried again, Duke Nukem Forever now launches and reaches the main menu.
Warhammer 40K Space Marine still saying servers are busy.

**Comment by PSebs** [comment]
GRID 2 boots but crashes after selecting "World Series Racing" or "GRID Online" in the title screen.

**Comment by tianon** [comment]
Unfortunately no dice for Star Trek (2013) (https://www.protondb.com/app/203250 / https://steamdb.info/app/203250/config/). :disappointed:

Play button switches, console shows that Proton launched and tried to run the game, then it just stops and we get a Play button again.

Is there an easy place I can look for a list of which of my own games uses CEG so I can help test more mainstream titles?  (This Star Trek title is the only one I've ever run into issues playing that got traced back to CEG and me subscribing to this issue, but I haven't tried _everything_ in my library.)

**Edit:** [steam-203250.log](https://github.com/ValveSoftware/Proton/files/7439003/steam-203250.log) :innocent:

**Edit 2x:** scratch all that; Star Trek is now working!!!  (Thanks to https://github.com/valvesoftware/proton/issues/753#issuecomment-955766280 for helping me realize my mistake.)

**Comment by serebit** [comment]
Call of Duty: Black Ops works totally fine now, minus one oddity that's also present if you attempt to run a cracked CEG-less EXE—the Ascension zombies map refuses to load. The game crashes if you try to start up that map.

**Comment by serebit** [comment]
Black Ops II sort of starts up, but it gives an Unhandled Exception error before the game window appears.

**Comment by Swizzler121** [comment]
Just tried Sleeping Dogs (original, not the re-release) and it fails with Servers too busy error code 2, I'll try it again later just in case the servers are actually busy, but all my other games are launching fine.

https://steamdb.info/app/202170/config/

I am running proton experimental on the latest steam beta

EDIT2: Closed/reopened my steam and it asked to do another beta update, after that update can now run the game, so i'm guessing even though it applied one update today I wasn't on the very-very latest!

![image](https://user-images.githubusercontent.com/36831696/139383165-b78f4c19-590b-47bf-8fae-a93481b2fea7.png)

**Comment by nmiell** [comment]
Space Marine -- creates a black window and then crashes, which I assume is an improvement over no CEG at all
Lara Croft & the Guardians of Light -- played through the first level without any problem (lots of stuttering from what I assume is shader compiling, though)

**If you're getting an error about the Steam servers being too busy, make sure you've switched to the _Proton Experimental_ compatibility tool, either in the global settings or in the per-game configuration. Proton 6.3 doesn't work.**

**Comment by lucifertdark** [comment]
I've tried Hitman Absolution, Lara Croft & the Guardian of Light & Lara Croft & the Temple of Osiris, all are working with the latest Proton Experimental. I'm going to test the others later, the real world is demanding my presence for a few hours today.

ps both Lara croft games have a stutter for me at the moment.

**Comment by mindinsomnia** [comment]
Just finished work for the day, so I was able to test this more extensively. Here are the outcomes on my laptop:

**Aliens vs Predator:** Worked on first attempt. he game launched in a tiny window, but I was able to increase the size and after that get into the first mission of the game without an issue.

**Duke Nukem Forever:** Game launches, splash screen 'flickers' while animating, game window opens and opening video begins playing. Two attempts to run game, both resulted in the game freezing shortly after launch before reaching the main menu.

**Sniper Elite V2:** Displayed launcher with Play button. Clicked play, a window briefly flashd on screen before disappearing. Two attempts, same result.

**Warhammer 40,000: Space Marine:** Game installed and attempted to launch, could see some logos, before the game crashed. Two attempts made, same outcome.

**Saints Row: The Third:** Game installed, ran, displayed video logos for a few moments, then crashed. Same result on second attempt.

**System Info:**
Operating System: Manjaro Linux
KDE Plasma Version: 5.22.5
KDE Frameworks Version: 5.87.0
Qt Version: 5.15.2
Kernel Version: 5.14.10-1-MANJARO (64-bit)
Graphics Platform: X11
Processors: 12 × Intel® Core™ i7-10750H CPU @ 2.60GHz
Memory: 15.5 GiB of RAM
Graphics Processor: Mesa Intel® UHD Graphics

**Comment by GloriousEggroll** [comment]
Lords of the Fallen works now. Played the intro level up to the first boss without any issues.

**Comment by shigutso** [comment]
Excellent news, thanks @Plagman !
Will test Typing of the Dead Overkill as soon as I finish work

**Comment by jm2** [comment]
Can confirm after opting into the Steam Beta channel and ensuring the latest updates are installed, I tested DiRT Showdown and GRID 2, both of which primarily have issues with CEG based on testing with stripped binaries. There's a new "Completing Installation..." step on first run followed by some steam client service windows (which interestingly look like cmd.exe with just a blinking cursor), which take about a minute to finish. For DiRT Showdown, it gets much closer then it did before, where it now shows the splash screen and a black window, but it crashes shortly afterward. GRID 2 on the other hand now successfully launches to the main menu, but crashes when loading the next sub-menu (possibly during the DLC check.) So there's definitely CEG-related progress being made now, which is promising compared to before.

**Comment by Devorlon** [comment]
Wow I installed Civ 5 an hour ago and have been looking for a fix.

Civ 5 now works, both the launcher and the game take a while to load (10s and 30s respectively) but I can start both a single and multiplayer game and complete a couple turns.

I'll try and report back if the singleplayer game I'm playing crashes in the long run, 

EDIT: Only one crash, but that was from repeated alt-tabing but iirc that's a issue on Windows as well. Use [gamescope](https://github.com/Plagman/gamescope) to solve the problem.

**Comment by HunterCZ122** [comment]
Mafia II (Classic) finally starts, but crashes when starting new game from menu. Doesn't happen with cracked executable, so it's CEG issue obviously.

**Comment by NoXPhasma** [comment]
> Lords of the Fallen works now. Played the intro level up to the first boss without any issues.

Hmm, that worked before. Are you sure that uses CEG? https://www.protondb.com/app/265300

I've tested Lara Croft and the Guardian of Light and that works now with Steam Beta + Proton Experimental. Haven't actually played it yet, but I get to the main menu.

**Comment by Fenteale** [comment]
> Call of Duty: Black Ops works totally fine now, minus one oddity that's also present if you attempt to run a cracked CEG-less EXE—the Ascension zombies map refuses to load. The game crashes if you try to start up that map.

IIRC This has been an issue forever even with just a completely normal windows install.  The Ascension dlc for some reason crashes if you don't have the Multiplayer component installed, try it again after installing the multiplayer.

Just tested the singleplayer campaign in Black Ops for myself. Everything worked great! I was even able to unlock achievements.

**Comment by Corben78** [comment]
[Aliens vs Predator (10680)](https://github.com/ValveSoftware/Proton/issues/530) seems to work now. I played through the trigger points that where known to me to cause issues, but it starts (takes a while, a small little window shows in the left top corner first, but then goes to full screen after a while) and seems playable. Will have to check tomorrow again, as I had this before, where it was working one day, but not the other. 

[Aliens: Colonial Marines (49540)](https://github.com/ValveSoftware/Proton/issues/3810) starts!

**Comment by xpander69** [comment]
[F.E.A.R 3 (21100)](https://github.com/ValveSoftware/Proton/issues/963) - launches until splash screen. Sits there quite a long time, executable will show up in the processes eventually and then quits. No window is created.
[steam-21100.log](https://github.com/ValveSoftware/Proton/files/7441629/steam-21100.log)

tried without esync, fsync, with wined3d, without hiding nvidia gpu etc. no dice.

**Comment by itsvic-dev** [comment]
my attempts with Saints Row: The Third have so far been unsuccessful - in my case, it gets stuck at "Completing installation... %", and won't get past that:

![image](https://user-images.githubusercontent.com/17727163/139431740-5182b136-bc5e-4b9f-855e-d2daaed10d90.png)

i'm forcing use of Proton Experimental:

![image](https://user-images.githubusercontent.com/17727163/139431851-9a9961a6-ec21-4020-a091-0c446a1eb144.png)

as native just crashes after the logo intros have finished playing

**Comment by MindrustUK** [comment]
@omaemae Make sure you have the latest steam beta client as well as proton experimental. I found this issue this morning before updating the client.

**Comment by itsvic-dev** [comment]
> @omaemae Make sure you have the latest steam beta client as well as proton experimental. I found this issue this morning before updating the client.

@MindrustUK i tried opting out and opting back into beta as it thought it was up-to-date, and...

![image](https://user-images.githubusercontent.com/17727163/139434096-b933781b-30ff-4f7b-8c77-537fe660f41a.png)

it works! i'm gonna play around for a bit to see what works and what doesn't, but i assume experience will be similar to playing a cracked version on Wine


**update:** works pretty well, i think performance even beats Wine in this case, however unpausing is still crashing the game

**update 2:** entering a mission also crashes, not sure if this happens on a cracked Wine version but i think it does as well

overall could be improved but it's pretty promising so far

**Comment by Probotector** [comment]
Sniper Elite V2 black window and then crash

Lara croft guardian of the light not start...

Saints Rows Third random crash in game

Tested in Arch Linux KDE stable with GTX 1070 and NVIDIA 495.44 stable driver

**Comment by Nervyalloy5** [comment]
[Call of Duty: Black Ops (42700)](https://github.com/ValveSoftware/Proton/issues/790) working as intended

[Call of Duty: Black Ops II (202970)](https://github.com/ValveSoftware/Proton/issues/808) Hits an unhandled exception on launch

[Call of Duty: Modern Warfare 2 (10180/10190)](https://github.com/ValveSoftware/Proton/issues/1037) working as intended

[Call of Duty: Modern Warfare 3 (42680)](https://github.com/ValveSoftware/Proton/issues/1265) working as intended

[Saints Row the Third (55230)](https://github.com/ValveSoftware/Proton/issues/2228) can load ingame but crashes in a few places
    Randomly during gameplay (tried dx9 mode and it seems better)
    Unpausing the game
    Clicking the load game option in the main menu
  Missions seem to work fine for me

Tested on Arch Linux, kernel 5.14.14-zen1-1-zen, Radeon RX 5700xt, Mesa 21.2.4-1.

**Comment by Shished** [comment]
Civilization V works - both DX 9 and 11 versions were tested. But Steam does some weird stuff before starting the game - it opens a window with text "Finishing installation ... %" and 2 or so wine cmd windows which closes before game starts.

**Comment by kisak-valve** [comment]
Hello everybody, friendly reminder that this issue report is for general CEG support. For now let's use this issue report for any odd behavior with the CEG sub-component in Steam for Linux.

Please use the separate compatibility reports for all game-specific feedback so that we can properly keep track of the current condition of the individual games. This includes pass/fail feedback on individual games. (One compatibility report per game title please.)

**Comment by Persephone-the-Sheep** [comment]
Steam client now spawns a window called steam client service happens about every 30 seconds and spins up my bluray drive and I loose control of what I'm doing, using Arch.  If there is a better place to report this please send me to it.

**Comment by serebit** [comment]
> Steam client now spawns a window called steam client service happens about every 30 seconds and spins up my bluray drive and I loose control of what I'm doing, using Arch. If there is a better place to report this please send me to it.

Same thing here, but only when installing a game or within a couple minutes after installing a game.

**Comment by thaewrapt** [comment]
Can confirm, those spawned terminal windows (lasting less than second though) happen to steal focus for that period, it also seems to create Proton prefixes for those freshly downloaded CEG-based games (before running them, right after the download) and complains about missing SteamAppID in the stderr of Steam client.

**Comment by Raklodder** [comment]
I got one of my CEG games running with experimental-bleeding-edge, but something seems to run in the background causing major frame loss 144>30 and stutter. It doesn't matter whether I reload the same level or tab in/out and my logfiles are huge.

**Comment by pchome** [comment]
AFAIK games like:
F.E.A.R. 3
Ghost Recon: Wildlands
Black Ops 2
Risen 2
maybe some other, require additional patch which was recently added to upstream Wine
https://github.com/wine-mirror/wine/commit/4db8cdc267efd62fc7349912fe352cf214c012f7
https://github.com/wine-mirror/wine/commit/75bccf99d1f90555c998f2c9b907ed9292ce47f3

**Comment by baldpale** [comment]
Related to #330 

I tried Mafia II again and unfortunately still doesn't work. It takes a long time to even launch the title screen (which is "better", since before the last patch it would just crash immediately), but as soon as I choose the story or DLC, it crashes.

It still loads and works fine after replacing mafia2.exe with a cracked version.

I use Steam beta client , Proton Experimental and Arch btw. Seems like it's not hardware related, because it's 100% reproducible and I tried on AMD PC and Intel laptop.

[steam_mafia2_out.txt](https://github.com/ValveSoftware/Proton/files/7447641/steam_mafia2_out.txt)

**Comment by mathew2214** [comment]
any plans to bring this fix to debian systems?
i know Valve really likes arch because Steam Deck but debian is a rather popular distro.

**Comment by kattjevfel** [comment]
@mathew2214 This shouldn't have anything to do with what distro you're running, just use the latest beta of Proton Experimental and you should be good to go.

**Comment by mathew2214** [comment]
@kattjevfel i am. no CEG games launch.
proton-experimental and steam client beta. 
as an example, here's my proton-log of trying to use BO1 (42700) along with my system info. 
https://gist.github.com/mathew2214/4925fe8bc52647fc3b6219a072a4252b

every CEG game behaves exactly like BO1. i have other games i can make logs for if need be.
yes, i deleted any pre-existing prefixes for these games before i realized CEG still just doesnt work. i even get the residual STEAMSTART files that always get left in the game folders as a dead giveaway of Valve's DRM not letting me use Valve's platform to play games i bought from Valve.

**Comment by kattjevfel** [comment]
> @kattjevfel i am. no CEG games launch. proton-experimental and steam client beta. as an example, here's my proton-log of trying to use BO1 (42700) along with my system info. https://gist.github.com/mathew2214/4925fe8bc52647fc3b6219a072a4252b

I see the issue right away, you're not using the Proton Experimental beta, as it says `Proton: 1635348752 experimental-6.3-20211027`. It should be saying something like `Proton: 1635544687 experimental-bleeding-edge-6.3-1836-20211029-paf96ce-w8a5bac-d1b1206-v430c77`, make sure you have enabled the branch at *Proton Experimental > Betas*.

**Comment by mathew2214** [comment]
![protonselector](https://user-images.githubusercontent.com/38667464/139596004-fcef1383-5b07-40db-a0b1-70033ab8fcce.png)

this is the closest to that option im given. i am relecutant to enable proton-experimental globally because im scared it might break other games that work fine on the current stable build.

**Comment by kattjevfel** [comment]
@mathew2214 you'll be wanting to look at the Proton Experimental entry, not the entry for the game.
![_2021-10-31_19-03-01](https://user-images.githubusercontent.com/6226312/139596149-df77fada-0c58-4abe-b7c5-13f6b0caec30.png)

**Comment by mathew2214** [comment]
@kattjevfel ah i see. 
guess im just out of luck then. 
Big Picture mode doesnt have a tools menu, and due to an unrelated [steam client bug](https://github.com/ValveSoftware/steam-for-linux/issues/7564), BPM is my only means of accessing a properties menu.

**Comment by mathew2214** [comment]
@kattjevfel that works. took me a while to figure out how to access a tool's properties in Big Picture mode. very unintuitive interface.

**Comment by NoXPhasma** [comment]
For me the wineserver process seems to have a memory leak with bleeding edge and CEG games. Not sure if this only affects CEG games, but those are the only ones I play with Experimental right now.  I've seen this happening with The Bureau and now with Homefront as well. It ate up to 13.7GB before I quit the game. 

https://gist.github.com/NoXPhasma/533c7562e7fb7366545517ca4a05ef67

![wineserver](https://user-images.githubusercontent.com/7719590/139602448-018f2730-cf54-4d25-8326-1965f3bbc6db.png)

**Comment by kattjevfel** [comment]
@NoXPhasma I guess this is the same as #5271, I didn't look at RAM when making that report.

**Comment by NoXPhasma** [comment]
Maybe, but I haven't noticed any unusual CPU load while that happened. However, I didn't ran into OOM, which I did before with The Bureau (luckily I run earlyoom) and would eventually when I wouln't exit Homefront.

**Comment by xpander69** [comment]
FEAR 3 works with latest bleeding edge proton experimental but the wine stuff is using loads of cpu
![ss_01112021_04 06 27](https://user-images.githubusercontent.com/1345116/139611839-9ae25bf7-0a38-438e-b368-a3334e4e71be.png)

edit:
ok disabling ESYNC/FSYNC makes the load more tolerable, but wineserver and tabtip still use quite a bit
![ss_01112021_04 21 15](https://user-images.githubusercontent.com/1345116/139612823-fc8c5bd1-cb44-4d99-a9e7-73194b89d5ad.png)

edit 2: yeah without ESYNC/FSYNC wineserver is slowly growing in memory use. like 100MB every 5 or 10 sec. got it until 14GB of RAM usage then i quit the game

**Comment by Patola** [comment]
Hypothesis: Proton Experimental Bleeding Edge's fsync version is the latest (PR done by André Almeida himself), but futex2 made for the previous fsync, used in linux-tkg, does not work with that fsync, so the 100% CPU usage bug occurs. Proton Exp. Bleeding Edge's fsync is supposed to be used with futex_waitv coming on Linux 5.16.

One wait to test this hypothesis would be, of course, trying a kernel with no fsync/futex2 compiled in. It should not activate fsync and should not have the 100% CPU usage bug.

**Comment by duderand** [comment]
FYI - When testing a different version of Proton with Grid 2 (44350) #1647 I noticed that removing the game compatdata folder made the game not start anymore. Verifying the local game files fixed the issue.

I used to remove the compatdata folder because it triggers the game setup (the setup that is run after download) and I think it better tests Proton integration.

**Comment by boltronics** [comment]
Tested:

* The Bureau: XCOM Declassified
* Warhammer 40,000: Space Marine - Anniversary Edition
* F.E.A.R. 3
* Mafia II (Classic)

These run flawlessly at a glance (got to the main menu, video, audio, mouse/keyboard inputs all seem fine). Very impressive. I've never been able to run these games before, and actually gave up a long time ago and ended up repurchasing them all on consoles. Most of these would predate Steam's current refund policies.

Steam Beta Update (Steam package versions: 1635476327, Build: Oct 29 2021, at 02:21:34)
Proton Experimental [bleeding-edge]
Debian 10 GNU/Linux,
Linux 5.14.0
Fury X (Mesa 21.2.1)
BenQ XL2730 2560x1440@144Hz with FreeSync.
Steam overlay reported a solid 144Hz max cap for all games to the main menu, which is as far as I tested.

**Comment by kisak-valve** [comment]
"Fixed a console window erroneously appearing while launching CEG games. (needs latest Proton Experimental)" in the [2021-11-02](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3076515048882667999) Steam client beta update should have improved the Steam client's behavior.

**Comment by duderand** [comment]
Has anyone noticed issues with games' network features (i.e. online gaming) or Steam integration like opening a DLC Store from the game? 

I have seen reports from other DRMs causing these kinds of issues and I have seen them with Grid 2 (44350) #1647 but I'm not sure if it's related to the DRM.

**Comment by brain-anti-freeze** [comment]
Happy to confirm that [Prototype 2 ](https://github.com/ValveSoftware/Proton/issues/1422) now runs great (I'm using the Proton [6.3-8 RC](https://github.com/ValveSoftware/Proton/issues/5307) with `WINEDLLOVERRIDES=”xaudio2_3=b”` set)

**Comment by brain-anti-freeze** [comment]
One thing I've noticed - I no longer get the 'Steam client service' console when launching games but I do see it when those games process their shaders sometimes.

**Comment by jkhsjdhjs** [comment]
> One thing I've noticed - I no longer get the 'Steam client service' console when launching games but I do see it when those games process their shaders sometimes.

Yes, the console window appearing on every game launch was a bug which was quickly fixed: https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/3076515048882667999

**Comment by GH-A-CC** [comment]
--- **Edit**:  Changing from Proton Experimental to a numbered version, restarting Steam and putting it back to Experimental in the global Steam settings (and then un-checking the compatibility option for the game itself set to Proton Experimental) then re-running it seems to have solved this.

When launching "Call of Duty: Modern Warfare 3 (Single Player)", I get a window with "Completing Installation" followed by a second pop-up with the windowtitle "Steam Client Service" showing the message "Failed to find Steam executable"

After a while (I would guess 30 seconds),  the error "The Steam servers are too busy to handle your request for Call of Duty: Modern Warfare 3. Error code (55)" is shown.

Proton Experimental,  Steam client build is 19th November 20:59:59 (Steam API v020,  Steam package versions: 1637358733)

**Comment by CSahajdacny** [comment]
Is there a list of Steam games with CEG?

**Comment by NoXPhasma** [comment]
@CSahajdacny Most probably not the entire list of CEG games, but it's a start https://gist.github.com/Plagman/eac7cf27885df24dec48f1477a447e1f

**Comment by kisak-valve** [comment]
Closing as fixed with the [2021-11-19](https://store.steampowered.com/oldnews/101607) Steam client update and [Proton 6.3-8](https://github.com/ValveSoftware/Proton/wiki/Changelog#63-8).

Follow up issues should be tracked separately and as usual, please use the existing compatibility reports for per-game feedback.
